src_features/signMessageEIP712/path.c

```diff
@@ -430,6 +442,8 @@ bool path_new_array_depth(const uint8_t *const data, uint8_t length) {
         return false;
     }
 
+    array_size = *data;
```

Comment:
I really feel like eth app would benefit from a lot more comments, I have trouble understanding what huge functions like this are performing both, on their overhaul effect and in the actual implementation :/